### PR TITLE
Apply new package name policies with fixed mixed-case list of packages

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/shared/utils.dart [hunk 1]
```diff
@@ -143,6 +143,18 @@ const List<String> _reservedWords = const <String>[
   'with'
 ];
 
+const knownMixedCasePackages = const [
```
Comment (kevmoo) [inline app/lib/shared/utils.dart]: Ran the tool

Known mixed case: Autolinker
Known mixed case: Babylon
Known mixed case: DartDemoCLI
Known mixed case: FileTeCouch
Unknown mixed case: Flutter_Nectar
Known mixed case: Google_Search_v2
Known mixed case: LoadingBox
Known mixed case: PolymerIntro
Known mixed case: Pong
Unknown mixed case: RAL
Known mixed case: Transmission_RPC
Unknown mixed case: ViAuthClient
Both mixed and lowercase: babylon

Thankfully, `Babylon` and `babylon` are owned by the same guy. He wants us to remove `Babylon`

**Comment by isoos** [comment]
> Should update the known list

Done.

File: app/lib/shared/utils.dart [hunk 2]
```diff
@@ -158,6 +173,12 @@ void validatePackageName(String name) {
   if (_reservedWords.contains(name.toLowerCase())) {
     throw new Exception('Package name must not be a reserved word in Dart.');
   }
+  final bool isLower = name == name.toLowerCase();
+  final bool matchesMixedCase =
+      knownMixedCasePackages.any((s) => s.toLowerCase() == name.toLowerCase());
+  if (!isLower && !matchesMixedCase) {
```
Comment (mkustermann) [inline app/lib/shared/utils.dart]: Why can't this be simply `if (!knownMixedCasePackages.contains(name)) { ... }` ?

File: app/lib/shared/utils.dart [hunk 3]
```diff
@@ -158,6 +173,12 @@ void validatePackageName(String name) {
   if (_reservedWords.contains(name.toLowerCase())) {
     throw new Exception('Package name must not be a reserved word in Dart.');
   }
+  final bool isLower = name == name.toLowerCase();
+  final bool matchesMixedCase =
+      knownMixedCasePackages.any((s) => s.toLowerCase() == name.toLowerCase());
```
Comment (mkustermann) [inline app/lib/shared/utils.dart]: This is doing the linear search even if `isLower == true` (which implies it's not in the `knownMixedCasePackages` list.

**Pull request merged**